server/verify.go

```diff
@@ -397,6 +398,7 @@ func parseMachineStateFromTPM(attestation *pb.Attestation, pcrs *tpmpb.PCRs, opt
 		return nil, fmt.Errorf("failed to validate the PCClient event log: %w", err)
 	}
 
+	// this shouldn't be here, otherwise it will be verified for every pcr banks redundantly
```

Comment:
I don't understand this comment. If you're saying we shouldn't call `verifyGceTechnology` and `parseTEEAttestation`, I agree. Add a TODO

Reply:
added, mainly `parseMachineStateFromTPM` is called for every pcr banks, which seems redundant to get TEE state from this function